bool get_firmware_version
----
string firmware_version